common/transport/amqp/src/amqp_common_errors.ts

```diff
@@ -110,6 +110,30 @@ export function translateError(message: string, amqpError: Error): AmqpTransport
     // In the case of a invalid Twin object, this will return the generic error message plus the specific error messaged provided from the server.
     let errorString = message + '. ' + (<any>amqpError.message).message.toString();
     error = new Error(errorString);
+  } else if (amqpError.constructor?.name) {
+    //
+    // Questions for reviewers:
+    //
+    // 1. We don't want to translate these errors because we already have the right error,
+    //    but, what do we do with the passed-in message?  Do we lose that info, or do we
+    //    have some sort of "inner exception". e.g. if the passed error is
+    //    "NotConnectedError: SenderLink disconnected", and message is "Could not send",
+    //    What do we want to return?
+    //
+    // 2. I added these 2 errors becauese I needed to in order to fix the bug. What about
+    //    the oher errors.X values above. like errors.ThrottlingError?  Should I also leave
+    //    those untranslated?
+    //
+    switch (amqpError.constructor.name) {
+      case 'ServiceUnavailableError':
+      case 'NotConnectedError':
+        error = amqpError;
+        amqpError = null;
+        break;
+      default:
+        error = new Error(message);
+        break;
+    }
```

Comment:
For the `ServiceUnavailableError` case, this just ends up in the `default` case. This whole thing would could just be an if-else.

I think you're aware of it and left this unfinished in case you need to deal with more error cases, but pointing it out just in case.

Reply:
> For the `ServiceUnavailableError` case, this just ends up in the `default` case. This whole thing would could just be an if-else.

is this true? because there's no break happening between 'ServiceUnavailableError' and 'NotConnectedError' they both will be handled by the no-translation case of 'NotConnectedError'. See: https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Statements/switch#methods_for_multi-criteria_case

Unless I'm misunderstanding the comment you made?

Reply:
@YoDaMa Suppose the value of `amqpError.constructor.name` is `'ServiceUnavailableError'`. Then it will match with the first case of the switch. There are no statements for it (and importantly, that means there is no `break`). Therefore, it will fall it end up in the `default` case.

Reply:
hrm but is your assumption of how switch cases correct? I could be wrong, but JS supports fallthrough behavior in the switches. Here's an excerpt from something: 

> JavaScript supports the fall-through behavior in switch. If a case statement does not have a break;, the code for the next cases are also executed until a break; is found. This is why the break; statement is crucial when using the switch statement. If the break; statement does not exist, all the cases after the "correct" case are executed until the last case, including the default case. This defeats the purpose of the switch statement as the statement is used when you have multiple outcomes and only want to pinpoint one outcome, not all of them.

Reply:
Yes, I realized my understanding of `switch` was wrong right after I sent that message. :)

Reply:
saw your edit :) cool! That being said, your confusion is justified, and I think linters can get mad about fallthrough.

See: https://eslint.org/docs/rules/no-fallthrough , which is enabled by default. Essentially it says "don't do this whole fallthrough thing, unless you tell us you're doing it with a comment", because it as error prone aspect of Javascript.